### PR TITLE
[MPOM-313] execute checkstyle in early phase of the build

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1170,6 +1170,7 @@ under the License.
         <executions>
           <execution>
             <id>checkstyle-check</id>
+            <phase>validate</phase>
```
Comment (michael-o) [inline pom.xml:1173]: This also means that autogenerated code won't be checked?

Comment (olamy) [reply pom.xml:1173]: correct so maybe generate-sources phase.
but do we really care about checkstyle generated code?

Comment (olamy) [reply pom.xml:1173]: it's just frustrating to run a long build and get a missing space at the end...
just forget a space in surefire-its it means you run everything for at least 1h (with a beefy laptop) and the build failure 😩

Comment (michael-o) [reply pom.xml:1173]: > correct so maybe generate-sources phase. but do we really care about checkstyle generated code? in this case we need to ask the plugin authors to change the style of the generated code?

Сorrect, we don't care,  I guess.

Comment (michael-o) [reply pom.xml:1173]: > it's just frustrating to run a long build and get a missing space at the end... just forget a space in surefire-its it means you run everything for at least 1h (with a beefy laptop) and the build failure 😩

Agree

**Review by slawekjaranowski — APPROVED** [review]

**Comment by slawekjaranowski** [comment]
Simple change - can save some of time during development.
Are there any objection before process it?

**Comment by olamy** [comment]
> Simple change - can save some of time during development. Are there any objection before process it?

not from me 🤣

**Comment by slawekjaranowski** [comment]
https://issues.apache.org/jira/browse/MPOM-313

**Review by slachiewicz — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #260

**Pull request merged**